src/Microsoft.ML.Api/DataViewConstructionUtils.cs

```diff
@@ -946,7 +946,17 @@ public override ValueGetter<TDst> GetGetter<TDst>()
             throw Contracts.ExceptNotImpl("Type '{0}' is not yet supported.", typeT.FullName);
         }
 
-        internal override Delegate GetGetterDelegate() => Utils.MarshalInvoke(GetGetter<int>, MetadataType.RawType);
+        internal override Delegate GetGetterDelegate()
+        {
+            Type genericClassType = typeof(MetadataInfo<T>);
+            MethodInfo baseMethodInfo = genericClassType.GetMethod("GetGetter");
+
+            Type[] genericArguments = new Type[] { MetadataType.RawType };
+            MethodInfo genericMethodInfo = baseMethodInfo.MakeGenericMethod(genericArguments);
+
+            object classInstance = Activator.CreateInstance(genericClassType, Kind, Value, MetadataType);
+            return genericMethodInfo.Invoke(classInstance, new object[] { }) as Delegate;
```

Comment:
Please don't do this... make a new private method `GetGetterDelegateCore` that is generic on the type and returns a delegate. That method will do nothing but return a `Delegate`. Use `MarshalInvoke` on that.

Just as a general rule, if you ever, *ever* think that the solution to a problem is hard coding the name of a method as a string constant, something has almost certainly gone wrong. ;)

Reply:
Thanks for the tip.

---
In reply to: [239576439](https://github.com/dotnet/machinelearning/pull/1836#discussion_r239576439) [](ancestors = 239576439)